<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>920</width>
    <height>512</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LineCounter</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../res.qrc">
    <normaloff>:/icons/res/main.ico</normaloff>:/icons/res/main.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,3">
    <property name="margin">
     <number>6</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="leftVBoxLayout">
      <item>
       <widget class="QGroupBox" name="dirsGroupBox">
        <property name="title">
         <string>Directories</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QTreeWidget" name="dirsTree">
           <property name="indentation">
            <number>0</number>
           </property>
           <attribute name="headerVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string/>
            </property>
           </column>
          </widget>
         </item>
         <item row="1" column="0">
          <layout class="QHBoxLayout" name="dirHorizontalLayout">
           <item>
            <spacer name="dirHorizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="addDirButton">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="removeDirButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Remove</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0">
          <widget class="QCheckBox" name="subdirsCheckBox">
           <property name="text">
            <string>Include subdirectories</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="filtersGroupBox">
        <property name="title">
         <string>Filters</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QTreeWidget" name="filtersTree">
           <property name="indentation">
            <number>0</number>
           </property>
           <attribute name="headerVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string/>
            </property>
           </column>
           <column>
            <property name="text">
             <string/>
            </property>
           </column>
          </widget>
         </item>
         <item row="1" column="0">
          <layout class="QHBoxLayout" name="filtersHBoxLayout">
           <item>
            <spacer name="filtersSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="newFilterButton">
             <property name="text">
              <string>New</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="editFilterButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="deleteFilterButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="exportGroupBox">
        <property name="title">
         <string>Export</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="1" column="0" colspan="2">
          <layout class="QHBoxLayout" name="exportHBoxLayout">
           <item>
            <spacer name="exportSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="exportSaveButton">
             <property name="text">
              <string>Save</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="exportFormatCombo"/>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="exportFormatLabel">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Format:</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="maximum">
         <number>0</number>
        </property>
        <property name="value">
         <number>-1</number>
        </property>
        <property name="textVisible">
         <bool>true</bool>
        </property>
        <property name="format">
         <string>%v/%m (%p%)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="startButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>50</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>START!</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>1</number>
      </property>
      <item>
       <widget class="QTreeWidget" name="statsTree">
        <property name="indentation">
         <number>0</number>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <attribute name="headerStretchLastSection">
         <bool>false</bool>
        </attribute>
        <column>
         <property name="text">
          <string>Filename</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Filesize</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Source lines</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Comment lines</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Empty lines</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Total lines</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <widget class="QTableWidget" name="statsTable">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>20</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="autoScroll">
         <bool>false</bool>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::NoSelection</enum>
        </property>
        <property name="horizontalScrollMode">
         <enum>QAbstractItemView::ScrollPerPixel</enum>
        </property>
        <attribute name="horizontalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderDefaultSectionSize">
         <number>20</number>
        </attribute>
        <attribute name="verticalHeaderMinimumSectionSize">
         <number>0</number>
        </attribute>
        <row>
         <property name="text">
          <string/>
         </property>
        </row>
        <column>
         <property name="text">
          <string/>
         </property>
        </column>
        <column>
         <property name="text">
          <string/>
         </property>
        </column>
        <column>
         <property name="text">
          <string/>
         </property>
        </column>
        <column>
         <property name="text">
          <string/>
         </property>
        </column>
        <column>
         <property name="text">
          <string/>
         </property>
        </column>
        <column>
         <property name="text">
          <string/>
         </property>
        </column>
        <item row="0" column="0">
         <property name="text">
          <string/>
         </property>
        </item>
        <item row="0" column="1">
         <property name="text">
          <string/>
         </property>
        </item>
        <item row="0" column="2">
         <property name="text">
          <string/>
         </property>
        </item>
        <item row="0" column="3">
         <property name="text">
          <string/>
         </property>
        </item>
        <item row="0" column="4">
         <property name="text">
          <string/>
         </property>
        </item>
        <item row="0" column="5">
         <property name="text">
          <string/>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="../res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>addDirButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>addDir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>116</y>
    </hint>
    <hint type="destinationlabel">
     <x>31</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeDirButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>removeDir()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newFilterButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>newFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>40</x>
     <y>168</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editFilterButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>editFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>281</y>
    </hint>
    <hint type="destinationlabel">
     <x>97</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteFilterButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>deleteFilter()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>236</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exportSaveButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveExport()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>372</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>start()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>444</y>
    </hint>
    <hint type="destinationlabel">
     <x>275</x>
     <y>405</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dirsTree</sender>
   <signal>currentItemChanged(QTreeWidgetItem*,QTreeWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateDirButtons(QTreeWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>64</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>filtersTree</sender>
   <signal>currentItemChanged(QTreeWidgetItem*,QTreeWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>updateFiltersButtons(QTreeWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>229</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>statsTree</sender>
   <signal>itemDoubleClicked(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>openFile(QTreeWidgetItem*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>338</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>361</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateDirButtons(QTreeWidgetItem*)</slot>
  <slot>openFile(QTreeWidgetItem*)</slot>
  <slot>updateFiltersButtons(QTreeWidgetItem*)</slot>
  <slot>addDir()</slot>
  <slot>removeDir()</slot>
  <slot>start()</slot>
  <slot>saveExport()</slot>
  <slot>newFilter()</slot>
  <slot>editFilter()</slot>
  <slot>deleteFilter()</slot>
 </slots>
</ui>
